pkg/controller/iosvariant/iosvariant_controller.go

```diff
@@ -126,55 +124,51 @@ func (r *ReconcileIOSVariant) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
-	if foundVariant != "" {
-		// we don't do a full reconciliation (update push app on UPS server based on CR content) but we
-		// only do initial creation of push apps.
-		reqLogger.Info("Skip reconcile: IOS Variant already exists in UPS", "IOSVaraint.Name", instance.Name)
-		return reconcile.Result{}, nil
+	if foundVariant.VariantId == "" {
```

Comment:
in the android variant controller, we are comparing the object. But here we compare the just the `VariantId` field. Should we keep them consistent?

Reply:
Yeah, good point. I think I wanted to compare the object here too, but for some reason in this case it wouldn't work. I'll try again, and if it doesn't work I'll change the others to be like this one.